src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/Zaimanhua.kt

```diff
@@ -225,16 +226,43 @@ class Zaimanhua : HttpSource(), ConfigurableSource {
     override fun searchMangaParse(response: Response): MangasPage =
         response.parseAs<ResponseDto<PageDto>>().data.toMangasPage()
 
+    override fun fetchSearchManga(
```

Comment:
Avoid the `fetch*` methods if possible. Try to replace with `searchMangaRequest` / `searchMangaParse`

Reply:
As the title says, this change is actually intended to display the ranking list in, not the search, so it doesn't use the same api as the search and returns a completely different data structure.
In the searchMangaRequest,query.isEmpty() && filters.filterIsInstance<RankingGroup>().firstOrNull()! = null can determine which api to use. But in searchMangaParse, I don't know how to tell if it's searching normally or getting the ranking list.

Reply:
I thought of ways to do it, but it would make the code even uglier.

Reply:
you forgot to remove the fetch method override, since the changes were made to search request and parse